src/test/regress/sql/gp_orphaned_files.sql

```diff
@@ -0,0 +1,209 @@
+-- s/ERROR:  Error on receive from seg\d+ slice\d+ \d+.\d+.\d+.\d+:\d+ pid=\d+: server closed the connection unexpectedly/ERROR:  Error on receive from segX sliceX X.X.X.X:X pid=X: server closed the connection unexpectedly/
+-- end_matchsubs
+
+-- Test case 1
+-- Check that orphaned files are not left on the coordinator and the standby
+-- when the files are created after checkpoint
+
+-- Create tables of different access methods and return command to check their
+-- files existence on the coordinator and the standby
+create or replace function createTables() returns text as
+$$
+declare
+  cmd text;
+begin
+  create table t_orphaned_h(i int)
+  distributed by (i);
+
+  create table t_orphaned_r(i int)
+  with (appendonly=true, orientation=row)
+  distributed by (i);
+  -- Create index to create block directory table
+  create index t_orphaned_r_i on t_orphaned_r(i);
+
+  create table t_orphaned_c(i int)
+  with (appendonly=true, orientation=column)
+  distributed by (i);
+  -- Create index to create block directory table
+  create index t_orphaned_c_i on t_orphaned_c(i);
+
+  -- Ensure that the mirrors have applied the filesystem changes
+  perform force_mirrors_to_catch_up();
+
+  -- The command do not output PGDATA directories to make it possible to run
+  -- the test without docker
+  select '\! ' ||
+         string_agg('cd ' || datadir || '&&' || lswc, ';' order by datadir)
+  into cmd
+  from (
+    select 'ls ' || string_agg(pg_relation_filepath(a.unnest), ' ')
+                 || ' 2>/dev/null | wc -l' lswc
+    from (
+      select unnest(array['t_orphaned_h'::regclass,
+                          't_orphaned_r'::regclass, 't_orphaned_r_i'::regclass,
+                          't_orphaned_c'::regclass, 't_orphaned_c_i'::regclass])
+      union all
+      select unnest(array[segrelid,
+                          blkdirrelid, blkdiridxid,
+                          visimaprelid, visimapidxid])
+        from pg_catalog.pg_appendonly
+       where relid in ('t_orphaned_r'::regclass, 't_orphaned_c'::regclass)
+    ) a
+  ) f,
+  (select datadir from gp_segment_configuration where content = -1) d;
+
+  return cmd;
+end
+$$ language plpgsql;
+
+checkpoint;
+
+-- Skip checkpoints on the coordinator
+select gp_inject_fault_infinite('checkpoint', 'skip', dbid)
+  from gp_segment_configuration
+ where role = 'p' and content = -1;
+
+-- Start transaction and create tables in it
+begin;
+select createTables() check_files
+\gset
+ 
+-- Make sure that the tables files exist on the coordinator and the standby
+:check_files
+
+-- Get segfault on the coordinator and reconnect after its restart
+select gp_inject_fault('before_read_command', 'segv', dbid)
+  from gp_segment_configuration
+ where role = 'p' and content = -1;
+
+-- The error message from psql can be different, so ignore it
+\! psql postgres -c "select 1" 2> /dev/null
+\! sleep 2
+\c regression
+
+-- All the inject faults has been reset after the coordinator restart
```

Comment:
minor grammar correction:
```suggestion
-- All the inject faults have been reset after the coordinator restart
```

Reply:
Fixed